node/rpc/src/chain_api.rs

```diff
@@ -150,7 +150,16 @@ where
 /// Prints hex-encoded representation of SHA-256 checksum and saves it to a file
 /// with the same name but with a `.sha256sum` extension.
 async fn save_checksum(source: &Path, hash: Output<Sha256>) -> Result<()> {
-    let encoded_hash = format!("{} {}", hash.encode_hex::<String>(), source.display());
+    let encoded_hash = format!(
+        "{} {}\n",
+        hash.encode_hex::<String>(),
+        source
+            .file_name()
+            .and_then(|i| i.to_str())
+            .ok_or_else(|| anyhow::Error::msg(
```

Comment:
will `anyhow!` macro work here?

Reply:
I'd use `.context("Failed to retrieve file name while saving checksum")`

Reply:
@aatifsyed The output of `and_then` is `Option`, I'm not sure if I can use `.context` to convert it to error

Reply:
@hanabi1224 `.context` works for both `Result` and `Option`.

Reply:
Fixed